packages/shared/src/components/modals/Prompt.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { ReactElement } from 'react';
+import { usePrompt } from '../../hooks/usePrompt';
+import { Button } from '../buttons/Button';
+import {
+  ConfirmationModal,
+  ConfirmationHeading,
+  ConfirmationDescription,
+  ConfirmationButtons,
+} from './ConfirmationModal';
+
+export function PromptElement(): ReactElement {
+  const { prompt } = usePrompt();
+  if (!prompt) return null;
+  const {
+    onFail,
+    onSuccess,
+    options: { title, description, cancelButton, okButton },
+  } = prompt;
+  return (
+    <ConfirmationModal isOpen onRequestClose={onFail}>
+      <ConfirmationHeading>{title}</ConfirmationHeading>
+      {!!description && (
+        <ConfirmationDescription>{description}</ConfirmationDescription>
+      )}
+      <ConfirmationButtons>
+        {cancelButton !== null && (
+          <Button
+            className="btn-secondary"
+            onClick={onFail}
+            {...(cancelButton || {})}
+          >
+            {cancelButton?.title ?? 'Cancel'}
+          </Button>
+        )}
```

Comment:
Maybe we should be less explicit here?
In the past we had some AB test where left and right actions swap around (not sure if they are still available)

Let me know what you think.

Reply:
I can add a `reverseButtons` parameter. WDYT?

Reply:
That sounds like a plan, however it might be the onClose header button still always needs the close action?

Reply:
You mean we should expose the close handler as well? I think we can do that if we need.

Reply:
Yep, let's tackle it if it comes to that.